web/source/dom/domEventHandlers.ts

```diff
@@ -641,69 +641,10 @@ namespace com.keyman.dom {
         target.scrollInput();
         // nextSibling - the scrollbar element.
       } else if(tTarg != scroller.nextSibling) { // Otherwise, if clicked on text in SPAN, set at touch position
-        var caret,cp,cpMin,cpMax,x,y,dy,yRow,iLoop;
-        caret=scroller.childNodes[1]; //caret span
-        cpMin=0;
-        cpMax=target.getText()._kmwLength();
-        cp=target.getTextCaret();
-        dy=document.body.scrollTop;
-
-        // Vertical scrolling
-        if(target.base instanceof target.base.ownerDocument.defaultView.HTMLTextAreaElement) {
-          yRow=Math.round(target.base.offsetHeight/(target.base as HTMLTextAreaElement).rows);
-          for(iLoop=0; iLoop<16; iLoop++)
-          {
-            y=dom.Utils.getAbsoluteY(caret)-dy;  //top of caret
-            if(y > touchY && cp > cpMin && cp != cpMax) {cpMax=cp; cp=Math.round((cp+cpMin)/2);}
-            else if(y < touchY-yRow && cp < cpMax && cp != cpMin) {cpMin=cp; cp=Math.round((cp+cpMax)/2);}
-            else break;
-            target.setTextCaret(cp);
-          }
-
-          while(dom.Utils.getAbsoluteY(caret)-dy > touchY && cp > cpMin) {
-            target.setTextCaret(--cp);
-          }
-
-          while(dom.Utils.getAbsoluteY(caret)-dy < touchY-yRow && cp < cpMax) {
-            target.setTextCaret(++cp);
-          }
-        }
-
-        // Caret repositioning for horizontal scrolling of RTL text
-
-        // snapOrder - 'snaps' the touch location in a manner corresponding to the 'ltr' vs 'rtl' orientation.
-        // Think of it as performing a floor() function, but the floor depends on the origin's direction.
-        var snapOrder;
-        if(target.dir == 'rtl') {  // I would use arrow functions, but IE doesn't like 'em.
-          snapOrder = function(a, b) {
-            return a < b;
-          };
-        } else {
-          snapOrder = function(a, b) {
-            return a > b;
-          };
-        }
-
-        for(iLoop=0; iLoop<16; iLoop++) {
-          x=dom.Utils.getAbsoluteX(caret);  //left of caret
-          if(snapOrder(x, touchX) && cp > cpMin && cp != cpMax) {
-            cpMax=cp;
-            cp=Math.round((cp+cpMin)/2);
-          } else if(!snapOrder(x, touchX) && cp < cpMax && cp != cpMin) {
-            cpMin=cp;
-            cp=Math.round((cp+cpMax)/2);
-          } else {
-            break;
-          }
-          target.setTextCaret(cp);
-        }
-
-        while(snapOrder(dom.Utils.getAbsoluteX(caret), touchX) && cp > cpMin) {
-          target.setTextCaret(--cp);
-        }
-        while(!snapOrder(dom.Utils.getAbsoluteX(caret), touchX) && cp < cpMax) {
-          target.setTextCaret(++cp);
-        }
+        let _this = this;
+        target.executeCaretSearchFunctor(function() {
+          _this.performCaretSearch(target, scroller, touchX, touchY);
+        });
```

Comment:
```suggestion
        target.executeCaretSearch(() => this.performCaretSearch(target, scroller, touchX, touchY));
```

Using arrow function means we have lexical `this`.